nixos/modules/system/boot/stage-2.nix

```diff
@@ -11,26 +11,28 @@ let
 
   useHostResolvConf = config.networking.resolvconf.enable && config.networking.useHostResolvConf;
 
-  bootStage2 = pkgs.substituteAll {
+  bootStage2 = pkgs.replaceVarsWith {
     src = ./stage-2-init.sh;
-    shellDebug = "${pkgs.bashInteractive}/bin/bash";
```

Comment:
This is dead code, right?

Reply:
Introduced here: https://github.com/NixOS/nixpkgs/commit/ee163c2c8012cdc5cdd8efce06a380e38e3018d7

Made ineffective here: https://github.com/NixOS/nixpkgs/commit/ac53b25f167746cff0885ee28c6a172a43b81984

Things like this are my rationale for desiring the checkPhase that blows up if the thing substituted isn't there. It's so easy to keep things entangled for literal decades when they're dead as a doornail.